src/processing/dom/server-dom-adapter.js

```diff
@@ -10,17 +10,70 @@ export default class ServerDomAdapter extends BaseDomAdapter {
         this.crossDomainPort = crossDomainPort;
     }
 
+    walk (el, processor) {
+        if (el.nodeName !== '#document' && el.nodeName !== '#text' && el.nodeName !== '#documentType' &&
+            el.nodeName !== '#comment')
+            processor(el);
+
+        if (el.childNodes)
+            el.childNodes.forEach(child => this.walk(child, processor));
+    }
+
+    remove (el) {
```

Comment:
AFAIK those methods are not in the Base adapter. I think we should add some kind of comment about those methods to avoid future confusions.